src/main/java/org/mule/extension/db/internal/domain/connection/type/resolver/CollectionTypeResolver.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.db.internal.domain.connection.type.resolver;
+
+import static org.mule.extension.db.internal.domain.connection.oracle.OracleConnectionUtils.getOwnerFrom;
+import static org.mule.extension.db.internal.domain.connection.oracle.OracleConnectionUtils.getTypeSimpleName;
+import static org.mule.extension.db.internal.domain.connection.oracle.OracleDbConnection.QUERY_OWNER_CONDITION;
+import org.mule.extension.db.internal.domain.connection.DefaultDbConnection;
+import org.mule.extension.db.internal.domain.type.ResolvedDbType;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Type resolver for array entities
+ * 
+ * @since 1.5.2
+ */
+public class CollectionTypeResolver implements TypeResolver {
```

Comment:
This class seems oracle specific. It should make that clear in the name and be in the oracle package. Also, I don't fully understand the value of this class. It does pretty much nothing but delegate in the connection object, so why isn't this logic contained in the connection object?

Reply:
The resolution of clob and blob data type is different for an array or a struct. And there are differences between an Oracle connection (which uses the data type name in a query) and a default connection (which uses the data type id and the data type name in the metadata). So we use double dispatch, and there is a type resolver for each data type needed that delegates some behavior to the connection object.